OpenRA.Mods.Cnc/Widgets/Logic/CampaignLogic/CampaignMissionPreviewLogic.cs

```diff
@@ -0,0 +1,125 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2014 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Widgets;
+using OpenRA.Mods.Common.Widgets.Logic;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Cnc.Widgets.Logic.CampaignLogic
+{
+	class CampaignMissionPreviewLogic
```

Comment:
This seems to be generalized already which is lovely (at least I dont see anything Cnc specific).